packages/client/src/components/ClientApp.svelte

```diff
@@ -46,11 +46,24 @@
   // Provide contexts
   setContext("sdk", SDK)
   setContext("component", writable({ id: null, ancestors: [] }))
-  setContext("context", createContextStore())
+
+  const globalContext = createContextStore()
+  setContext("context", globalContext)
 
   let dataLoaded = false
   let permissionError = false
   let embedNoScreens = false
+  let cachedActiveScreenId
+
+  // Clear component context entries from the global context
+  // when the active page changes.
+  const purgeGlobalComponentContext = screenId => {
+    if (!screenId || cachedActiveScreenId == screenId) return
+    cachedActiveScreenId = screenId
+    globalContext.actions.clearComponentContext()
+  }
+
+  $: purgeGlobalComponentContext($screenStore?.activeScreen?._id)
```

Comment:
I understand the purpose of this, but I think we should come up with a cleaner way of handling it than trying to whitelist all properties that we want to persist between screen changes. I have a few ideas in mind... For now I think it would be better to take this out of this PR as it's a separate issue, and we can tackle it elsewhere.